python/llm/example/Text-Generation-WebUI/README.md

```diff
@@ -0,0 +1,132 @@
+
+This tutorial provides a step-by-step guide on how to use Text-Generation-WebUI to run Hugging Face transformers-based applications on BigDL-LLM.
+
+## 1. Prepare the environment
+
+Please use a python environment management tool (we recommend using Conda) to create a python enviroment and install necessary libs.
+
+### 1.1 Install Conda
+
+For Linux users, please open a terminal and run below commands:
+
+```
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash ./Miniconda3-latest-Linux-x86_64.sh
+conda init
+```
+
+### 1.2 Create Environment
+
+Create a Python 3.9 environment with the name you choose
+
+```
+conda create -n bigdl-llm python=3.9
+```
+
+Then activate the environment
+```
+conda activate bigdl-llm
+```
+
+
+### 1.3 Install BigDL-LLM
+
+BigDL-LLM Backend WebUI has already support running applications on both CPU and GPU device, please prepare the environment according to your device and requirements.
+
+```bash
+# on cpu device
+pip install --pre --upgrade bigdl-llm[all]
+
+# on gpu device
+pip install --pre --upgrade bigdl-llm[xpu] -f https://developer.intel.com/ipex-whl-stable-xpu
+```
+
+### 1.4 Install other required dependencies
+
+```bash
+pip install -r requirements.txt
+```
+Note: Text-Generation-WebUI requires transformers version >= 4.36.0
+
+### 1.5 Best Known Configuration on Linux
+For optimal performance on Intel GPUs, it is recommended to set several environment variables:
+```bash
+# configure oneAPI environment variables
+source /opt/intel/oneapi/setvars.sh
+
+export USE_XETLA=OFF
+export SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1
+```
+
+
+## 2. Start the WebUI Server
+
+### 2.1 For INT4 Optimizations
+
+For a quick start, you may run the script as below to start WebUI directly, it will automatically optimize and accelerate LLMs using INT4 optimizations.
+```bash
+python server.py
+```
+
+### 2.2 Optimizations for Other Percisions
+
+To enable optimizations for more precisions (`sym_int4`, `asym_int4`, `sym_int8`, `fp4`, `fp8`, `fp16`, `mixed_fp4`, `mixed_fp8`, etc.), you may run the command as below:
+```bash
+python server.py --load-in-low-bit
+```
+
+### 2.3 Access the WebUI
+
+After the successful startup of the WebUI server, it will provide links to access the WebUI as below. Please open the public URL in your browser to access the full functionality of the WebUI.
+
+```bash
+Running on local URL:  http://127.0.0.1:7860
+Running on public URL: https://644acd77d9cd7c5909.gradio.live
```

Comment:
shall we hide the secret token here?

Reply:
I think it should be generated randomly.